cmake/ExternalPalace.cmake

```diff
@@ -6,8 +6,10 @@
 #
 
 # Force build order
-set(PALACE_DEPENDENCIES mfem libCEED)
+set(PALACE_DEPENDENCIES mfem)
+
 if(PALACE_BUILD_EXTERNAL_DEPS)
+  list(APPEND PALACE_DEPENDENCIES libceed)
```

Comment:
@hughcars I know we didn't want `PALACE_WITH_SPACK`, but I don't know how to work around this at the CMake level without introducing a new variable here.

- Non spack builds want to build libCEED
- Spack builds don't want to build libCEED